tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -19,13 +19,54 @@
 #include <ROOT/RNTupleMerger.hxx>
 #include <ROOT/RNTupleModel.hxx>
 #include <ROOT/RNTupleUtil.hxx>
+#include <ROOT/RPageStorageFile.hxx>
+#include "TFile.h"
 
 Long64_t ROOT::Experimental::RNTuple::Merge(TCollection *inputs, TFileMergeInfo *mergeInfo)
 {
-   if (inputs == nullptr || mergeInfo == nullptr) {
+   // Check the inputs
+   if (!inputs || inputs->GetEntries() < 3 || !mergeInfo)
       return -1;
+
+   // Input parameters
+   std::string ntupleName;
+   std::unique_ptr<Internal::RPageSinkFile> destination;
+   std::vector<std::unique_ptr<Internal::RPageSourceFile>> sources;
+   std::vector<Internal::RPageSource *> sourcePtrs;
+
+   // Parse the inputs
+   Long64_t counter{0};
+   TIter itr(inputs);
+   while (const auto &pitr = itr()) {
+      if (counter == 0) {
+         ntupleName = std::string(pitr->GetName());
+      } else if (counter == 1) {
+         TFile *outFile = pitr->InheritsFrom("TFile") ? dynamic_cast<TFile *>(pitr) : nullptr;
+         if (!outFile)
+            return -1;
+         RNTupleWriteOptions writeOpts;
+         writeOpts.SetUseBufferedWrite(false);
+         destination = std::make_unique<Internal::RPageSinkFile>(ntupleName, *outFile, writeOpts);
+      } else {
+         TFile *inFile = pitr->InheritsFrom("TFile") ? dynamic_cast<TFile *>(pitr) : nullptr;
+         RNTuple *anchor = inFile ? inFile->Get<RNTuple>(ntupleName.c_str()) : nullptr;
+         if (!anchor)
+            return -1;
+         sources.push_back(Internal::RPageSourceFile::CreateFromAnchor(*anchor));
+      }
+      counter++;
+   }
```

Comment:
Instead of using the `counter` construction, I think it may be clearer to just handle the first two entries as special cases:
```suggestion
   TIter itr(inputs);
   std::string ntupleName = itr()->GetName();

   TFile *outFile = dynamic_cast<TFile *>(itr());
   if (!outFile)
      return -1; 
   RNTupleWriteOptions writeOpts;
   writeOpts.SetUseBufferedWrite(false);
   auto destination = std::make_unique<Internal::RPageSinkFile>(ntupleName, *outFile, writeOpts);

   std::vector<std::unique_ptr<Internal::RPageSourceFile>> sources;
   std::vector<Internal::RPageSource *> sourcePtrs;

   while (const auto &pitr = itr()) {
      TFile *inFile = dynamic_cast<TFile *>(pitr);
      RNTuple *anchor = inFile ? inFile->Get<RNTuple>(ntupleName.c_str()) : nullptr;
      if (!anchor)
         return -1; 
      sources.push_back(Internal::RPageSourceFile::CreateFromAnchor(*anchor));
   }
```

Reply:
Thanks Jonas. I adopted this.